File: LeetCode 227-Basic calculater II/basic_calculator_II.go
package main

import "fmt"

func calculate(s string) int {
	finalResult, tempResult, operand := 0, 0, 0
	operator := '+'

	for idx, char := range s {
		if '0' <= char && char <= '9' {
			operand = operand*10 + int(char-'0')
		}

		if idx == len(s)-1 ||
			char == '+' ||
			char == '-' ||
			char == '*' ||
			char == '/' {
			if operator == '-' || operator == '+' {
				finalResult += tempResult
				tempResult = operand
				if operator == '-' {
					tempResult = -operand
				}
			} else if operator == '*' {
				tempResult *= operand
			} else if operator == '/' {
				tempResult /= operand
			}

			operand = 0
			operator = char
		}
	}

	finalResult += tempResult

	return finalResult
}

func main() {
	fmt.Println("result:", calculate("10 + 2*3 - 6/3"))
}

/*
output:
result: 14
*/

File: LeetCode 227-Basic calculater II/basic_calculator_II.py
class Solution(object):
    def calculate(self, s):
        """
        :type s: str
        :rtype: int
        """
        finalResult, tempResult, operand = 0, 0, 0
        operator = "+"
        
        for idx, char in enumerate(s):
            if '0' <= char <= '9':
                operand = operand * 10 + int(char)
            if idx == len(s) - 1 or char in "+-*/":
                if operator in "+-":
                    finalResult += tempResult
                    tempResult = operand if operator == "+" else -operand
                elif operator == "*":
                    tempResult *= operand
                elif operator == "/":
                    if tempResult < 0:
                        tempResult = -1 * (abs(tempResult)/operand)
                    else:
                        tempResult /= operand
            
                operand = 0
                operator = char
                
        
        finalResult += tempResult
        
        return finalResult

if __name__ == "__main__":
    solution = Solution()
    print("result", solution.calculate("10 + 2*3 - 6/3"))

"""
output:
('result', 14)
"""